File: grabber-desktop/src/api/stig.ts
import { invoke } from "@tauri-apps/api/core";

export interface StigFinding {
  v_id: string;
  title: string;
  severity: string;
  fedramp_req_ids: string[];
  status: string;
  actionable: boolean;
  expected_value: string;
  actual_value: string;
  details: string;
  needs_text_input: boolean;
  remediation: string[];
}

export interface StigApplyRequestDto {
  credential_id: string;
  tenant_name: string;
  v_ids: string[];
  text_input?: string;
  output_dir: string;
}

export interface StigRemediationOutcome {
  v_id: string;
  target: string;
  label: string;
  detail: string;
  log_path?: string;
}

export const stigScan = (credentialId: string) =>
  invoke<StigFinding[]>("stig_scan", { credentialId });

export const startStigRemediation = (request: StigApplyRequestDto) =>
  invoke<StigRemediationOutcome[]>("start_stig_remediation", { request });

File: grabber-desktop/src-tauri/src/commands/stig.rs
use std::path::PathBuf;

use tauri::State;
use the_grabber::engine::{StigApplyRequest, StigFinding, StigRemediationOutcome};

use crate::dto::StigApplyRequestDto;
use crate::error::GuiError;
use crate::state::AppState;

/// Evaluate every Okta STIG check against the tenant behind the credential.
#[tauri::command]
pub async fn stig_scan(
    credential_id: String,
    state: State<'_, AppState>,
) -> Result<Vec<StigFinding>, GuiError> {
    let engine = state.engine.clone();
    Ok(engine.stig_scan(&credential_id).await?)
}

/// Apply remediation for the selected findings and append to the remediation log.
#[tauri::command]
pub async fn start_stig_remediation(
    request: StigApplyRequestDto,
    state: State<'_, AppState>,
) -> Result<Vec<StigRemediationOutcome>, GuiError> {
    if request.v_ids.is_empty() {
        return Err(GuiError::Validation(
            "Select at least one finding to remediate".into(),
        ));
    }

    let engine = state.engine.clone();
    let req = StigApplyRequest {
        credential_id: request.credential_id,
        tenant_name: request.tenant_name,
        v_ids: request.v_ids,
        text_input: request.text_input,
        output_dir: PathBuf::from(request.output_dir),
    };
    Ok(engine.stig_apply(req).await?)
}
